include/dlaf/matrix/tile.h

```diff
@@ -180,78 +248,44 @@ class Tile<const T, device> {
   // e.g. in dataflow, .then, ...
   Tile(hpx::shared_future<ConstTileType> tile, const SubTileSpec& spec)
       : Tile<const T, device>(tile.get(), spec) {
-    sfc_ = std::move(tile);
+    dep_tracker_ = std::move(tile);
   }
 
-  TileElementSize size_;
-  memory::MemoryView<ElementType, device> memory_view_;
-  SizeType ld_;
-
-  // With C++17 the following objects can be joined in a variant.
-  std::unique_ptr<promise_t<TileType>> p_;
-  hpx::shared_future<TileType> sf_;
-  hpx::shared_future<ConstTileType> sfc_;
+  TileDataType data_;
+  std::variant<std::monostate, TilePromise, hpx::shared_future<TileType>,
+               hpx::shared_future<ConstTileType>>
+      dep_tracker_;
 };
 
-template <class T, Device device>
-Tile<const T, device>::Tile(const TileElementSize& size,
-                            memory::MemoryView<ElementType, device>&& memory_view, SizeType ld) noexcept
-    : size_(size), memory_view_(std::move(memory_view)), ld_(ld) {
-  DLAF_ASSERT(size_.isValid(), size_);
-  DLAF_ASSERT(ld_ >= std::max<SizeType>(1, size_.rows()), ld, size_.rows());
-  DLAF_ASSERT(size_.isEmpty() || linearSize(size_, ld_) <= memory_view_.size(), size_, ld_,
-              memory_view_.size());
-}
-
-template <class T, Device device>
-Tile<const T, device>::Tile(Tile&& rhs) noexcept
-    : size_(rhs.size_), memory_view_(std::move(rhs.memory_view_)), ld_(rhs.ld_), p_(std::move(rhs.p_)),
-      sf_(std::move(rhs.sf_)), sfc_(std::move(rhs.sfc_)) {
-  rhs.setDefaultSize();
-}
-
 template <class T, Device device>
 Tile<const T, device>::~Tile() {
-  if (p_) {
+  if (std::holds_alternative<TilePromise>(dep_tracker_)) {
+    auto& p_ = std::get<TilePromise>(dep_tracker_);
     if (std::uncaught_exceptions() > 0)
-      p_->set_exception(std::make_exception_ptr(ContinuationException{}));
+      p_.set_exception(std::make_exception_ptr(ContinuationException{}));
     else
-      p_->set_value(Tile<ElementType, device>(size_, std::move(memory_view_), ld_));
+      p_.set_value(std::move(this->data_));
   }
```

Comment:
```suggestion
  }
  else
  {
    DLAF_ASSERT(std::holds_alternative<std::monostate>(dep_tracker_));
  }
```
or are all the other variants allowed here?

Reply:
Everything is allowed:
monostate <=> no dependent tile has to be set
promise <=> the promise ha to be set to unlock the pipeline
shared_future <=> this is a subtile and the destruction of the shared_future will unlock the pipeline